File: C_Programming/Practise_Programs/PP_161.c
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_161
//
//  Description       : This program displays the hexadecimal value
//                      of a bit mask for each bit position from
//                      1 to 32 using the left shift operator.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 20/08/2026
//
//  Time Complexity   : O(1)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

#include<stdio.h>

typedef unsigned int UNIT;

int main()
{
    UNIT iMask = 1;
    int iCnt = 0;

    for(iCnt = 1; iCnt <= 32; iCnt++)
    {
        printf("%d : %X\n", iCnt, iMask);  
        iMask = iMask << 1;
    }   
    
    return 0;
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      No input required.
//
//  Sample Output :
//
//      1 : 1
//      2 : 2
//      3 : 4
//      4 : 8
//      5 : 10
//      ...
//      32 : 80000000
//
//////////////////////////////////////////////////////////////////

File: Java_Programming/Practise_Programs/PP_161.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_161
//
//  Description       : This program accepts a number from the
//                      user and displays a pattern of stars and
//                      hash symbols. Three consecutive positions
//                      display a star followed by three positions
//                      displaying a hash symbol.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 04/09/2026
//
//  Time Complexity   : O(n)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.*;

class PP_161
{
    public static void Display(int iNo)
    {
        int iCnt = 0;

        for(iCnt = 1; iCnt <= iNo; iCnt++)
        {
            if(((iCnt - 1) / 3) % 2 == 0)
            {
                System.out.print("*\t");
            }
            else
            {
                System.out.print("#\t");
            }
        }

    }
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;

        System.out.println("Enter the elements : ");
        iValue = sobj.nextInt();

        Display(iValue);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input :
//      Enter the elements : 10
//
//  Sample Output :
//
//      *	*	*	#	#	#	*	*	*	#
//
//////////////////////////////////////////////////////////////////